pkg/logs/sender/sender.go

```diff
@@ -24,16 +24,19 @@ type Strategy interface {
 type Sender struct {
 	inputChan    chan *message.Message
 	outputChan   chan *message.Message
+	hasError     chan bool
 	destinations *client.Destinations
 	strategy     Strategy
 	done         chan struct{}
+	lastError    error
 }
 
 // NewSender returns a new sender.
 func NewSender(inputChan chan *message.Message, outputChan chan *message.Message, destinations *client.Destinations, strategy Strategy) *Sender {
 	return &Sender{
 		inputChan:    inputChan,
 		outputChan:   outputChan,
+		hasError:     make(chan bool),
```

Comment:
nit: I wonder if we can simplify `hasError` and `lastError` by just making is a single error channel (`make(chan error)`) instead of a boolean + error.

Reply:
We have to know if the error state has changed (from having an error vs not having an error) so we have to remember what the last state was somewhere. Also I kinda figured passing a nil error trough the channel to indicate "no error" was not as clear as a boolean.